packages/transport/src/http.js

```diff
@@ -19,9 +19,10 @@ import * as API from '@ucanto/interface'
  * @param {URL} options.url
  * @param {(url:string, init:API.HTTPRequest) => API.Await<FetchResponse>} [options.fetch]
  * @param {string} [options.method]
+ * @param {Headers | Map<string, string> | Record<string, string>} [options.headers]
```

Comment:
I would just use `Record<string, string>` as this is the existing type for the `headers` that can be passed to the `request()` method - i.e. as defined by `HTTPRequest`. This simplifies merging significantly.